pandas/tests/dtypes/test_dtypes.py

```diff
@@ -451,26 +451,41 @@ def test_hash_vs_equality(self):
         assert hash(dtype2) == hash(dtype2)
         assert hash(dtype2) == hash(dtype3)
 
-    def test_construction(self):
-        with pytest.raises(ValueError):
-            IntervalDtype('xx')
-
-        for s in ['interval[int64]', 'Interval[int64]', 'int64']:
-            i = IntervalDtype(s)
-            assert i.subtype == np.dtype('int64')
-            assert is_interval_dtype(i)
-
-    def test_construction_generic(self):
-        # generic
-        i = IntervalDtype('interval')
-        assert i.subtype == ''
+    @pytest.mark.parametrize('subtype', [
+        'interval[int64]', 'Interval[int64]', 'int64', np.dtype('int64')])
+    def test_construction(self, subtype):
+        i = IntervalDtype(subtype)
+        assert i.subtype == np.dtype('int64')
         assert is_interval_dtype(i)
-        assert str(i) == 'interval[]'
 
-        i = IntervalDtype()
+    @pytest.mark.parametrize('subtype', [None, 'interval', 'interval[]'])
```

Comment:
``interval[]`` prob doesn't make sense, is there a reason to keep it?

Reply:
Yeah, just put that there for legacy purposes, since it was previously a valid representation:

```python
In [4]: IntervalDtype('interval')
Out[4]: interval[]
```
Agree that it doesn't really make sense, and can remove it.

Reply:
removed